packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -110,7 +114,7 @@ export function getValueForAttribute(
       return expected === undefined ? undefined : null;
     }
     const value = node.getAttribute(name);
-    if (value === '' + (expected: any)) {
+    if (value === toStringOrTrustedType(getToStringValue(expected))) {
```

Comment:
This call site seems like it breaks with trusted types already.

But this fix doesn't seem right. Why call `getToStringValue`? Why would you need to ignore `function` values or `symbol`? `getAttribute` should never return those types.

`toStringOrTrustedType` also doesn't seem like the right fix neither.

For trusted types, the server generated string won't match the object on the client, so this wouldn't end up matching. Causing a hydration error. This will need to dig into the trusted type and inspect the value to see if it's equivalent.

For `REACT_OPAQUE_OBJECT_TYPE` the returned object will never match. So that will also be a hydration error.

It seems to me that this needs to just perform some more custom matching logic instead of trying to use the `toStringOrTrustedType` helper.